pkg/controller/operators/catalog/subscription/reconciler.go

```diff
@@ -182,7 +182,18 @@ func (c *catalogHealthReconciler) health(now *metav1.Time, catalog *v1alpha1.Cat
 // healthy returns true if the given catalog is healthy, false otherwise, and any error encountered
 // while checking the catalog's registry server.
 func (c *catalogHealthReconciler) healthy(catalog *v1alpha1.CatalogSource) (bool, error) {
-	return c.registryReconcilerFactory.ReconcilerForSource(catalog).CheckRegistryServer(catalog)
+	if catalog.Status.Reason != "" {
```

Comment:
We only set Reason on a catalog if it’s unhealthy?

Reply:
Good catch, I meant to swap this for a specific value check.